monix-execution/shared/src/main/scala/monix/execution/misc/Local.scala

```diff
@@ -64,42 +59,26 @@ object Local {
   def clearContext(): Unit =
     localContext.set(defaultContext())
 
-  /** Execute a synchronous block of code without propagating any `Local.Context`
+  /** Execute a  block of code without propagating any `Local.Context`
     * changes outside.
     */
-  def isolate[R](f: => R): R =
-    macro Macros.isolate
-
-  /** Execute an asynchronous block of code without propagating any `Local.Context`
-    * changes outside.
-    */
-  def isolate[R](f: => Future[R]): Future[R] = {
-    // TODO: rewrite as macro
-    val prev = Local.getContext()
-    val current = Local.getContext().mkIsolated
-    Local.setContext(current)
-
-    try {
-      FutureUtils.transform[R, R](f, result => {
-        Local.setContext(prev)
-        result
-      })(TrampolineExecutionContext.immediate)
-    } finally {
-      Local.setContext(prev)
-    }
+  def isolate[R: CanIsolate](f: => R): R = {
+    CanIsolate[R].isolate(f)
   }
 
   /** Execute a block of code using the specified state of
     * `Local.Context` and restore the current state when complete.
     */
-  def bind[R](ctx: Context)(f: => R): R =
-    macro Macros.localLet
+  def bind[R: CanIsolate](ctx: Context)(f: => R): R = {
```

Comment:
I don't know why mima complains about this change but not for `isolate`

Reply:
It complains because this is an actual method, whereas `isolate` is a macro.

Changing `isolate` right now will not change.

This means that this `bind` needs to be deprecated. Can you try the hard source deprecation trick via `private[Local]`?

Reply:
Not sure what's the trick but I think I managed to get rid of binary incompatibility without exposing old methods